src/docfx/build/Build.cs

```diff
@@ -29,21 +30,36 @@ private static List<Document> GlobFiles(Context context, Docset docset)
                            .ToList();
         }
 
-        private static Task BuildFiles(Context context, List<Document> files)
+        private static Task BuildFiles(Context context, List<Document> files, TableOfContentsMap tocMap)
         {
-            return ParallelUtility.ForEach(files, file => BuildOneFile(context, file));
+            return ParallelUtility.ForEach(files, file => BuildOneFile(context, file, tocMap));
+        }
+
+        private static async Task<TableOfContentsMap> BuildTocMap(Context context, List<Document> files)
+        {
+            var builder = new TableOfContentsMapBuilder();
+
+            var tocFiles = files.Where(f => f.ContentType == ContentType.TableOfContents);
+            if (!tocFiles.Any())
```

Comment:
iterate twice

Reply:
Or can also remove this check

---
In reply to: [184586582](https://github.com/dotnet/docfx/pull/2696#discussion_r184586582) [](ancestors = 184586582)